packages/amplify-graphql-api-construct/src/internal/authorization-modes.ts

```diff
@@ -5,19 +5,61 @@ import { IRole, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
 import {
   AuthorizationModes,
   ApiKeyAuthorizationConfig,
-  IAMAuthorizationConfig,
   LambdaAuthorizationConfig,
   OIDCAuthorizationConfig,
   UserPoolAuthorizationConfig,
 } from '../types';
 
 type AuthorizationConfigMode =
-  | (IAMAuthorizationConfig & { type: 'AWS_IAM' })
+  | { type: 'AWS_IAM' }
```

Comment:
Why are we removing the `IAMAuthorizationConfig` type?

Reply:
Because the `type AuthorizationConfigMode` is only used locally in this file for type checking below and non of `IAMAuthorizationConfig` properties are used in these lines (which was confusing when I was reading this code first time).
https://github.com/aws-amplify/amplify-category-api/blob/11336ca59d035e5b57b62e11c583158291c9b8b4/packages/amplify-graphql-api-construct/src/internal/authorization-modes.ts#L36-L37